pkg/skaffold/cluster/minikube.go

```diff
@@ -162,18 +129,15 @@ func matchServerURLFor(kubeContext string, serverURL *url.URL) (bool, error) {
 		return false, fmt.Errorf("failed to unmarshal data: %w", err)
 	}
 
-	for _, v := range data.Valid {
-		if v.Config.Name != kubeContext {
-			continue
-		}
+	serverURL, err := url.Parse(server)
+	if err != nil {
+		logrus.Tracef("invalid server url: %v", err)
+	}
 
-		if v.Config.Driver != HyperKit && v.Config.Driver != VirtualBox {
-			// Since node IPs don't match server API for other drivers we assume profile name match is enough.
-			// TODO: Revisit once https://github.com/kubernetes/minikube/issues/6642 is fixed
-			return true, nil
-		}
+	for _, v := range data.Valid {
 		for _, n := range v.Config.Nodes {
-			if serverURL.Host == fmt.Sprintf("%s:%d", n.IP, n.Port) {
+			if err == nil && serverURL.Host == fmt.Sprintf("%s:%d", n.IP, n.Port) {
+				// TODO: Revisit once https://github.com/kubernetes/minikube/issues/6642 is fixed
 				return true, nil
 			}
 		}
```

Comment:
I think we still want the profileId == kubeContext heuristic:
```golang
 		if v.Config.Name != kubeContext {
 		 		logrus.Infof("Found minikube profile with same name as %q", kubeContext)
 		 		return true, nil
 		}
```

Reply:
but by your own example: having GKE, kind and minikube clusters all called test, this heuristic would fail. Even though there's a workaround for that, it could break folks on release and as such doesn't seem that backwards compatible. Maybe it'd be better if minikube fixed https://github.com/kubernetes/minikube/issues/6642 so we wouldn't have to go around guessing.